lib/puppet/provider/package/gem.rb

```diff
@@ -48,11 +48,14 @@ def self.provider_command
   #
   # In this case, causing the puppet_gem provider to inherit the parent gem provider's convenience gemcmd() methods, with the wrong path.
 
-  def self.execute_gem_command(command, command_options)
+  def self.execute_gem_command(command, command_options,
+                               custom_environment = {})
```

Comment:
Just a cosmetically suggestion, since the method is not so long, we can keep the signature on a single line: 
```ruby
def self.execute_gem_command(command, command_options, custom_environment = {})
```